app/controllers/doorkeeper/tokens_controller.rb

```diff
@@ -18,12 +18,16 @@ def revoke
       # Doorkeeper does not use the token_type_hint logic described in the
       # RFC 7009 due to the refresh token implementation that is a field in
       # the access token model.
-      revoke_token if authorized?
 
-      # The authorization server responds with HTTP status code 200 if the token
-      # has been revoked successfully or if the client submitted an invalid
-      # token
-      render json: {}, status: 200
+      if authorized?
+        revoke_token
+        # The authorization server responds with HTTP status code 200 if the token
+        # has been revoked successfully or if the client submitted an invalid
+        # token
+        render json: {}, status: 200
+      else
+        render json: { error: :unauthorized_client }, status: :forbidden
```

Comment:
I think it will be good to have one more OAuth error response field, `error_description`, like:

https://github.com/doorkeeper-gem/doorkeeper/blob/72425150a95c4c5874ac2926b0ede90e705bd736/lib/doorkeeper/oauth/error.rb#L6-L12